src/plugins/discover/public/components/common/error_alert.tsx

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import { EuiButton, EuiEmptyPrompt } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { useHistory } from 'react-router-dom';
+
+export const DiscoverError = ({ error }: { error: Error }) => {
+  const history = useHistory();
+
+  const goToMain = () => {
+    history.push('/');
+  };
+
+  return (
+    <EuiEmptyPrompt
+      paddingSize="l"
+      iconType="alert"
+      iconColor="danger"
+      title={
+        <h2>
+          {i18n.translate('discover.discoverError.title', {
+            defaultMessage: 'Error loading Discover',
+          })}
+        </h2>
+      }
+      body={<p>{error.message}</p>}
+      actions={
+        <EuiButton color="primary" fill onClick={goToMain}>
+          <FormattedMessage id="discover.goToMainButtonText" defaultMessage="Go to main" />
+        </EuiButton>
```

Comment:
I think this button makes not much sense when this error happens on 'main', and I don't think there's a good alternative for it like "Please reload your browser", "Please call support", "Please come back a little later and hope it's fine", so in this case, unless someone has a better idea, I wouldn't show a button. 

But were it makes sense if e.g. it happens on another route than 'main'. But there's one problem with the wording. "main" is our internal wording, so I'm adding @gchaps here. So the case would be, there's a fundamental error in the part of Discover that doesn't show the data table or document explorer. How could we name the link to the "main" view of Discover? Many thx!

Reply:
To move forward here, and since it's already better then before (the wrong error message was displayed). I'd suggest it simple:

**Title**: `Cannot load this page`
**Description**: the message of the error (error.message) - this can be technical but we'd need this info if there's an error in this case
**Button: `Go to Discover main view` - but just displayed when it's not an error of the main view 

Why not **Go back** or something? We are implementing a new route for Discover which is a link from external, so the user might not have bin in Discover before this route